src/routes/safes/handlers/estimations.rs

```diff
@@ -6,48 +6,101 @@ use crate::common::models::backend::transactions::{
 };
 use crate::common::models::page::Page;
 use crate::providers::info::{DefaultInfoProvider, InfoProvider};
-use crate::routes::safes::models::{SafeTransactionEstimation, SafeTransactionEstimationRequest};
+use crate::routes::safes::models::{
+    SafeTransactionEstimation, SafeTransactionEstimationRequest, SafeTransactionEstimationV2,
+};
 use crate::utils::context::RequestContext;
 use crate::utils::errors::ApiResult;
 use crate::utils::http_client::{HttpClient, Request};
 use std::cmp::max;
 
-pub async fn estimate_safe_tx_gas(
+async fn get_last_known_nonce_and_estimate(
     context: &RequestContext,
     chain_id: &str,
     safe_address: &str,
     safe_transaction_estimation_request: &SafeTransactionEstimationRequest,
-) -> ApiResult<SafeTransactionEstimation> {
+) -> ApiResult<(String, u64, Option<u64>)> {
     let info_provider = DefaultInfoProvider::new(chain_id, &context);
-    let estimation_url = core_uri!(
-        info_provider,
-        "/v1/safes/{}/multisig-transactions/estimations/",
-        safe_address
-    )?;
+
+    let current_nonce = info_provider.safe_info(safe_address).await?.nonce;
+
     let latest_multisig_tx_url = core_uri!(
```

Comment:
Shouldn't this be inside a provider?

Reply:
The provider returns the whole info object, but we only need the nonce here

Reply:
Maybe, I followed the pattern from the existing code